terraform/examples/offline.tf

```diff
@@ -0,0 +1,761 @@
+# This file is meant to define the 'offline' environment, which lives in the 'crash' trust zone.
+# See also https://github.com/zinfra/backend-issues/wiki/trust-zone-environments (TODO: move that page over to either the zinfra/backend-wiki's wiki, or to zinfra/backend-wiki's repository)
+
+# What is an Offline VPC?
+# The only host reachable from the outside is a bastion host. It has internet access, and provides SSH services. It can access all of the other hosts in the VPC via SSH only.
+
+# To deploy this file, you will need a user with the policies "AmazonEC2FullAccess", "IAMFullAccess", "AmazonS3FullAccess", and "AmazonDynamoDBFullAccess".
+# FUTUREWORK: drill down on the above.
+
+terraform {
+  required_version = ">= 0.12.0"
+
+  backend "s3" {
+    encrypt = true
+    region  = "eu-central-1"
+
+    # TODO: create IAM policy which only allows access to this bucket under
+    # envrionments/crash and to the -crash dynamodb table
+    bucket = "z-terraform-remote-state"
+
+    key = "environments/offline/bootstrap"
+
+    dynamodb_table = "z-terraform-state-lock-dynamo-lock-environment-offline"
+  }
+}
+
+# In AWS, (eu-central-1)
+provider "aws" {
+  region = "eu-central-1"
+}
+
+# Used for the in-VPC EC2 endpoint.
+data "aws_security_group" "default" {
+  name   = "default"
+  vpc_id = module.vpc.vpc_id
+}
+
+module "vpc" {
+  source = "github.com/terraform-aws-modules/terraform-aws-vpc"
+
+  name = "offline"
+
+  cidr = "172.17.0.0/20"
+
+  azs             = ["eu-central-1a", "eu-central-1b", "eu-central-1c"]
+  private_subnets = ["172.17.0.0/22", "172.17.4.0/22", "172.17.8.0/22"]
+  public_subnets  = ["172.17.12.0/24", "172.17.13.0/24", "172.17.14.0/24"]
+
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  enable_dhcp_options      = true
+  dhcp_options_domain_name = "offline.zinfra.io"
+#  dhcp_options_domain_name_servers = 
+  
+  # In case we run terraform from within the environment.
+  # VPC endpoint for DynamoDB
+  enable_dynamodb_endpoint = true
+
+  # In case we run terraform from within the environment.
+  # VPC Endpoint for EC2
+  enable_ec2_endpoint              = true
+  ec2_endpoint_private_dns_enabled = true
+  ec2_endpoint_security_group_ids  = [data.aws_security_group.default.id]
+
+  enable_nat_gateway = true
+  one_nat_gateway_per_az = false
+# Use this only in productionish environments.
+#  one_nat_gateway_per_az = true
+
+  tags = {
+    Owner       = "Backend Team"
+    Environment = "Offline"
+    TrustZone   = "Crash"
+  }
+  vpc_tags = {
+    Owner       = "Backend Team"
+    Name        = "vpc-offline"
+    TrustZone   = "Crash"
+  }
+}
```

Comment:
This block (from "provider aws" until the end of vpc) should be moved to `terraform/modules/aws_vpc`. A few things need to be put into variables. So the domain name, the environment, and the trustZone are all good candidates for variables. Declaring and using variables is described [on this page](https://www.terraform.io/docs/configuration/variables.html). I wouldn't recommend to over-do variables at this point, so in a first stage I think we can keep the subnets and the availability zones hard-coded. (Or we use variables with the existing values as a [default](https://www.terraform.io/docs/configuration/variables.html#declaring-an-input-variable))